packages/hash/api/src/model/knowledge/user.model.ts

```diff
@@ -385,19 +404,73 @@ export default class extends EntityModel {
   }
 
   async joinOrg(
-    _graphApi: GraphApi,
-    _params: { org: any; responsibility: string; updatedByAccountId: string },
+    graphApi: GraphApi,
+    params: {
+      org: OrgModel;
+      responsibility: string;
+    },
   ) {
-    /** @todo: re-implement this method */
-    throw new Error("user.joinOrg is not yet re-implemented");
+    const { org, responsibility } = params;
+
+    const orgMembership = await OrgMembershipModel.createOrgMembership(
+      graphApi,
+      { responsibility, org },
+    );
+
+    await this.createOutgoingLink(graphApi, {
+      linkTypeModel: WORKSPACE_TYPES.linkType.ofOrg,
+      targetEntityModel: orgMembership,
+    });
+  }
+
+  async getOrgMemberships(graphApi: GraphApi): Promise<OrgMembershipModel[]> {
+    const { data: outgoingOrgMembershipLinks } = await graphApi.getLinksByQuery(
+      {
+        all: [
+          {
+            eq: [{ path: ["source", "id"] }, { literal: this.entityId }],
+          },
+          {
+            eq: [
+              { path: ["type", "versionedUri"] },
+              {
+                literal: WORKSPACE_TYPES.linkType.ofOrg.schema.$id,
+              },
+            ],
+          },
+        ],
+      },
+    );
+
+    return await Promise.all(
+      outgoingOrgMembershipLinks.map(async ({ targetEntityId }) => {
+        const orgMembership = await OrgMembershipModel.getOrgMembershipById(
+          graphApi,
+          {
+            entityId: targetEntityId,
+          },
+        );
+
+        if (!orgMembership) {
+          throw new Error("Critical: could not find target of link");
+        }
+
+        return orgMembership;
+      }),
+    );
   }
 
   async isMemberOfOrg(
-    _graphApi: GraphApi,
-    _params: { orgEntityId: string },
+    graphApi: GraphApi,
+    params: { orgEntityId: string },
   ): Promise<boolean> {
-    /** @todo: re-implement this method */
-    throw new Error("user.isMemberOfOrg is not yet re-implemented");
+    const orgMemberships = await this.getOrgMemberships(graphApi);
+
+    const orgs = await Promise.all(
+      orgMemberships.map((orgMembership) => orgMembership.getOrg(graphApi)),
+    );
+
+    return !!orgs.find(({ entityId }) => entityId === params.orgEntityId);
```

Comment:
Is this conventional? I don't write as much JS/TS nowadays as I should, but this double exclamation mark feels pretty gross and kinda too magical.

Reply:
I find it useful, but understand especially when you're coming from other languages that it feels weird. Would you prefer:
```suggestion
    return orgs.find(({ entityId }) => entityId === params.orgEntityId) !== undefined;
```
(in which case just commit the suggestion, I don't feel strongly either way)

Reply:
Spent a while looking for convention and seeing if people view this as a codesmell. There are a lot of examples of people suggesting to use `Boolean()` instead, but apparently that can cause problems in things like `filter` in TypeScript.

I'm fine with us leaving this as is. Although I will say that your suggestion is different from the code `!== undefined` isn't the same as `!!`.

Reply:
> Although I will say that your suggestion is different from the code !== undefined isn't the same as !!

It's different, yes - but it's also correct. `.find()` will always return `Org` or `undefined`, so `!== undefined` is sufficient to check if the array item was found.

Reply:
Which is why I think I'd generally err on the side of using that. Because it's being clear about the possible options, and would mean we'd have to revisit this if we changed the types.

The problem with `!!` and similar in my mind is that if we introduced return values that could be falsey, we'd probably get silent errors because TS thinks it's completely fine to turn an empty array or a `0` into a false value, even if we were only trying to exclude undefined things. But I appreciate that doesn't matter here because we're just using the inbuilt `find` and that's not going to change (probably)